### ISSUE TITLE
Group by with round brackets

### ISSUE THREAD

**Issue by marcelomachado** [RubenVerborgh/SPARQL.js#127]
The following code just parse and unparse an sparql query:

```
const sparqljs = require("sparqljs")
const parser = new sparqljs.Parser();
const generator = new sparqljs.Generator;

let parsedQuery = parser.parse('SELECT ?x { ?x ?y ?z. } group by ?x');
console.log(parsedQuery);

let generatedQuery = generator.stringify(parsedQuery);
console.log(generatedQuery); // returns:  SELECT ?x WHERE { ?x ?y ?z. }\nGROUP BY (?x)
```

The brackets are always added in the group by clause because the `SparqlGenerator.js` (line 55) contains:
```var result = isString(it.expression) ? it.expression : '(' + this.toExpression(it.expression) + ')';```

None of the cases that I tested `it.expression` were `string`. However, in [SPARQL 1.1 specification](https://www.w3.org/TR/sparql11-query/#groupby) there is no brackets in the group by clause, unless in alias or manipulation cases.

For some triplestores, such as Apache Jena, the brackets are not a problem, however, for some others are (e.g. AllegroGraph).

**Comment by RubenVerborgh** [comment]
Confirmed to be a bug: https://www.w3.org/TR/sparql11-query/#rGroupCondition

**Comment by QuarksToQuasars** [comment]
Hi @RubenVerborgh, [metaphacts](https://github.com/metaphacts) contributed this fix and we would highly appreciate if it would be included in and distributed through a new sparql.js release. Do you have any plans/schedule for the next release or other dependencies? Just asking for planning purpose. Thanks

**Comment by RubenVerborgh** [comment]
Thanks, released v3.4.3.
